README.md

```diff
@@ -0,0 +1,210 @@
+- **Explanation**: An `else` block after an `if` which contains a `return` is allowed. Technically the `else` is redundant (and Airbnb disallows it), but this rule allows for more readable control flow.
+- **Example**:
+
+        // Correct pattern
+        if (fooString === 'foo') {
+            return 30;
+        } else {
+            return 50;
+        }
+
+####[`no-invalid-this`](http://eslint.org/docs/rules/no-invalid-this)
+- **Setting**: `"error"`
+- **Explanation**: Do not use `this` outside of classes. Using strict mode, `this` can be undefined in these cases.
+- **Example**: See [the ESLint `no-invalid-this` docs](http://eslint.org/docs/rules/no-invalid-this) for examples of allowed and disallowed behavior.
+
+####[`no-shadow`](http://eslint.org/docs/rules/no-shadow)
+- **Setting**: `"error"`
+- **Explanation**: Do not redefine variables in a child scope that are already declared in the upper scope.
+- **Example**:
+
+        // Linter error
+        var foo = 'bar';
+        if (fizz) {
+            var foo = 'buzz';
+        }
+
+####[`object-curly-spacing`](http://eslint.org/docs/rules/object-curly-spacing)
+- **Setting**: `["error", "never"]`
+- **Explanation**: Do not insert extra spaces inside of curly brackets.
+- **Example**:
+
+        // Correct pattern
+        var obj = {foo: 42}
+
+        // Linter error
+        var obj2 = { foo: 42 }
+
+####[`one-var`](http://eslint.org/docs/rules/one-var)
+- **Setting**: `["error", "always"]`
+- **Explanation**: Only use one variable declaration per function. This reinforces awareness of ES5's block scoping and hoisting, which can reduce bugs.
+- **Example**:
+
+        // Correct pattern
+        function() {
+            var foo, bar, baz,
+                fizz = 'buzz',
+                foz = 49;
+        }
+
+        // Linter error
+        function() {
+            var fizz = 'buzz';
+
+            if (foo) {
+                var baz = 'bar';
+            }
+        }
+
+####[`one-var-declaration-per-line`](http://eslint.org/docs/rules/one-var-declaration-per-line)
+- **Setting**: `["error", "initializations"]`
+- **Explanation**: Variables that are not initialized can be declared on the same line. Only one initialization is allowed per line.
+- **Example**:
+
+        // Correct pattern
+        var foo, bar,
+            fizz = 'buzz';
+
+        // Linter error
+        var foo, bar, fizz = 'buzz';
+
+####[`quote-props`](http://eslint.org/docs/rules/quote-props)
+- **Setting**: `["error", "consistent"]`
+- **Explanation**: If no properties of an object need to be wrapped in quotes (i.e., they are all camelCased strings), then they may remain unwrapped by quotes. If any object property requires quotes, then wrap all object properties in quotes regardless of their individual need for it.
+- **Example**:
+
+        // Correct patterns
+        var foo = {
+            bar: 'buzz',
+            biff: 'bop'
+        }
+
+        var foo2 = {
+            'bar': 'buzz',
```

Comment:
I'm going to change our rule to `consistent-as-needed` because that's what I actually documented here mistakenly so I'll assume that that's what the people who already thumbsedup wanted when they reviewed.